src/Truncate.js

```diff
@@ -0,0 +1,41 @@
+import styled from 'styled-components'
+import {variant, maxWidth} from 'styled-system'
+import PropTypes from 'prop-types'
+import {TYPOGRAPHY, COMMON} from './constants'
+import theme from './theme'
+
+const truncateVariants = variant({
+  variants: {
+    target: {
+      display: 'inline-block',
+      verticalAlign: 'top'
+    }
+  }
+})
+
+const Truncate = styled('div')`
+  ${TYPOGRAPHY}
+  ${COMMON}
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  ${truncateVariants}
+  ${maxWidth}
+`
+
+Truncate.defaultProps = {
+  as: 'div',
+  maxWidth: 125,
+  theme
+}
+
+Truncate.propTypes = {
+  ...TYPOGRAPHY.propTypes,
+  ...COMMON.propTypes,
+  maxWidth: PropTypes.number,
```

Comment:
styled-system will take a lot of different propTypes for maxWidth, but it looks like it's predominantly using numbers. I can update that, if you'd like it to be more specific. https://styled-system.com/api#layout